### PR TITLE
e2e: move LogFailedContainers out of e2e test framework util.go

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Move `LogFailedContainers` out of e2e test framework `util.go`.
To resolve circular references, this PR also handles `rc_util.go`.

/cc @oomichi @MorrisLaw

**Which issue(s) this PR fixes**:

Ref #84615

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/framework/rc/rc_utils.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ import (
 	clientset "k8s.io/client-go/kubernetes"
 	scaleclient "k8s.io/client-go/scale"
 	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2ekubectl "k8s.io/kubernetes/test/e2e/framework/kubectl"
 	testutils "k8s.io/kubernetes/test/utils"
 )
```
Comment (oomichi) [inline test/e2e/framework/rc/rc_utils.go]: We need to rename `RcByNameContainer` for solving the following lint error:

`test/e2e/framework/rc/rc_utils.go:35:6: func name will be used as rc.RcByNameContainer by other packages, and that stutters; consider calling this ByNameContainer`

**Review by oomichi — COMMENTED** [review]
The change itself seems good for me because this doesn't add any invalid dependencies and good separation from the core framework.
Just one comment about a lint error.

File: test/e2e/framework/rc/rc_utils.go (hunk 1 above)
Comment (SataQiu) [reply test/e2e/framework/rc/rc_utils.go]: Thanks @oomichi 
Updated.

**Review by oomichi — COMMENTED** [review]
Thanks for updating

/lgtm
/approve

**Comment by MorrisLaw** [comment]
Looking good @SataQiu , I'm not sure why it failed. I re-ran it again, maybe it's a flakey test?

**Comment by dims** [comment]
/milestone v1.18

looks like we missed the boat. let's try after code thaw

**Comment by SataQiu** [comment]
@MorrisLaw  Yes, I'm also confused why the test didn't pass. 
/test pull-kubernetes-node-e2e-containerd

**Comment by SataQiu** [comment]
@oomichi  Need LGTM again :)

**Comment by hwdef** [comment]
/lgtm 
Thanks for your contribute!

**Pull request merged**